### PR TITLE
107 refactor indexenum class

### PR DESCRIPTION
**Pull request by samvanderzwan**
This pull request can be reviewed and merged if the other outstanding pull request have been handled. A rebase might then be required to avoid merge conflicts

### CODE REVIEW

**Comment by samvanderzwan** [comment]
Rebased this branch on master, now it can be reviewed

File: ci/win32/profiling.cmd
Comment (MichielTukker) [inline ci/win32/profiling.cmd:1]: what does this have to do with "refactoring indexenum" class??

File: doc/solver/assets/solver_producer.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.. _solver-producer:
+
+Solver producer class
```
Comment (MichielTukker) [inline doc/solver/assets/solver_producer.rst:3]: The name deviates from the names used in code  (Solver.ProductionAsset).  As this is dev-documentation, it's important to keep this in sync!

File: src/simulator_core/entities/assets/asset_abstract.py [hunk 2]
```diff
@@ -44,19 +44,24 @@ class AssetAbstract(ABC):
...
         self.from_junction = None
         self.to_junction: Junction = None
         self.name: str = asset_name
         self.asset_id: str = asset_id
         self.output: List[Dict[str, float]] = []
 
+    def __repr__(self) -> str:
+        """Method to print string with the name of the asset."""
+        return self.__class__.__name__ + " " + self.name
+
```
Comment (MichielTukker) [inline src/simulator_core/entities/assets/asset_abstract.py:64]: Is this ever used anywhere?? (not used in the current unit tests)

**Review by MichielTukker — CHANGES_REQUESTED** [review]
I don't see any changes related to an indexenum class?? 

Is the gh changes list   messed up due to merge commits (thus showing changes/commits from other branches?)

Maybe this PR needs to be renamed/updated?

**Comment by MichielTukker** [comment]
Branch commit history is f**** up due to incomplete (but committed!!) merge/rebase?   I did a manual merge in a separate branch and will create a new PR

**Pull request closed without merging**